[Sean Iau Yang] iP

seaniy opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

seaniy committed:
added Duke.java with Level-1 increment

seaniy committed:
updated Duke.java with Level-2 increment

seaniy committed:
Level-3

seaniy committed:
Level-4

seaniy committed:
A-TextUiTesting

seaniy committed:
Level-5

seaniy committed:
Level-6

seaniy committed:
Level-7: Added Save List Functionality

seaniy committed:
Level-8: Add Parsing of LocalDate for Deadlines

seaniy committed:
Merge branch 'branch-Level-8'

seaniy committed:
Fix bug when parsing local date from file

seaniy committed:
A-MoreOOP: Apply OOP

seaniy committed:
Add duke package

seaniy committed:
Add JUnit tests for Ui and Parser classes

seaniy committed:
Update ParserTest.java

seaniy committed:
Revert "Update ParserTest.java"

This reverts commit 11b792d8c732b5e52ce6f225fee0f591d7c3ee3e.

seaniy committed:
Revert "Revert "Update ParserTest.java""

This reverts commit f45b1058ec927dc94c4f27b657055395d04f24fb.

seaniy committed:
Merge branch 'add-gradle-support' into master

seaniy committed:
Add Gradle

seaniy committed:
Add Java Documentation

Tomashiwa commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate date;
```
Should this be protected or private?

Tomashiwa commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,116 @@
+                }
+            } catch(DukeException error) {
+                ui.showInputError();
+            }
```
Perhaps you can consider extracting all these into their individual Command class?

Tomashiwa commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,15 @@
+package duke;
+
+public class Event extends Task {
+    protected String timing;
```
Should this be protected or private?

Tomashiwa commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+
+import java.time.LocalDate;
+
+public class Parser {
```
Maybe you can consider converting this into a utility class whereby all the parse functions are static?

Tomashiwa commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+import java.time.LocalDate;
+
+public class Parser {
+    private final Ui ui = new Ui();
```
Isn't this ui unused?

Tomashiwa commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+
+public class Storage {
+    protected String filePath;
```
Should this be protected or private?

Tomashiwa commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+
+public class Storage {
+    protected String filePath;
+    private static final Ui ui = new Ui();
```
Perhaps you can just pass in the ui instance you already created in Duke?

Tomashiwa commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.List;
+
+public class TaskList {
+    protected List<Task> tasks;
```
Should this be protected or private?

Tomashiwa commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.util.Scanner;
+
+public class Ui {
```
Similar to Parser class, maybe you can consider converting this to a utility class with static functions?

Tomashiwa commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    public void showMessage(String message) {
+        System.out.println(divider + message + "\n" + divider);
```
Maybe you can consider using the faster System.out.printf with formatted string?

Tomashiwa commented on src/test/java/duke/UiTest.java:
```diff
@@ -0,0 +1,37 @@
+    public void setUp() {
+        System.setOut(new PrintStream(outputStreamCaptor));
+    }
+
```
Should the test be just calling the function? Maybe you can call functions that will invoke the ui showing those messages, then see if your ui actuall respond.

Tomashiwa reviewed (commented):
- Should probably add some JavaDoc to the classes and functions
- Some variables can just be private since the class is not a parent class
- ui and parser class can be converted into utility classes with static functions, so you no longer need to create an instance of them in multiple classes

frisciliasultan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String timing;
+
+    public Event(String description, String location) {
```
Perhaps location is supposed to be timing?

frisciliasultan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,57 @@
+    private final Ui ui = new Ui();
+
+    public Command parseCommand(String input) throws DukeException {
+        String[] inputBreakdown = input.split(" ", 2);
```
Should it be in plural form since it is an array?

frisciliasultan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+                "Type \"help\" to view the list of available commands.");
+    }
+
+    public void showGoodBye() {
```
Is it showGoodbye or showGoodBye?

frisciliasultan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,25 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
Perhaps add javadoc comments for public methods?

frisciliasultan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,15 @@
+
+    public Event(String description, String location) {
+        super(description);
+        this.timing = timing;
```
Perhaps timing should be date so that it is more consistent with the deadline task?

seaniy committed:
Add find task functionality

seaniy committed:
Update files to meet coding standards

seaniy committed:
Merge branch 'branch-A-CodingStandard'

seaniy committed:
Merge branch 'branch-Level-9'

seaniy committed:
Update JavaDoc

seaniy committed:
Added Gui

seaniy committed:
Update MainWindow.fxml

Fix bug where text in dialog box gets cut off if its too long

seaniy committed:
Update build.gradle

seaniy committed:
Add assertions to check for errors in code

seaniy committed:
Check code quality

seaniy committed:
Merge pull request #2 from seaniy/branch-A-Assertions

Add assertions to check for errors in code

seaniy committed:
Merge branch 'master' into branch-A-CodeQuality

seaniy committed:
Merge updated master to branch-A-CodeQuality

seaniy committed:
Merge pull request #3 from seaniy/branch-A-CodeQuality

Check code quality

seaniy committed:
Update search function

Add more flexibility to search function to allow for partial searches as well as task category searches

seaniy committed:
Update Better GUI

seaniy committed:
Update some exception handling

seaniy committed:
Update GUI

seaniy committed:
Add a User Guide

seaniy committed:
Fix DialogBox Size Issue

seaniy committed:
Merge branch 'branch-A-UserGuide'

seaniy committed:
Set theme jekyll-theme-cayman

seaniy committed:
Update minor bugs

Change package reference for build.gradle

Fix minor bug in doc README

seaniy committed:
Merge branch 'master' of https://github.com/seaniy/ip

seaniy committed:
Update README.md

seaniy committed:
Set theme jekyll-theme-midnight

seaniy committed:
Update README.md

seaniy committed:
Update JUnit Testing

seaniy committed:
Update code quality and exceptions
